### PR TITLE
OD-361 [Feature] Display files in the Files field with full metadata and interactivity

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://weboo.atlassian.net/browse/OD-361

## Description
Changed file field layout.

## Screenshots/screencasts
https://share.getcloudapp.com/Jru49ANG

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko @AndrRyaz

### CODE REVIEW

File: css/form.css [hunk 1]
```diff
@@ -716,11 +712,43 @@ html[dir="rtl"] .radio.radio-icon input[type="radio"] + label > span.check {
...
+  max-width: calc(100% - 30px);
+}
+
+.file-holder.editable .file-icon {
+  width: 30px;
+  height: 30px;
+}
+
+.file-holder.editable .file-icon .fa {
+  color:red;
```
Comment (tonytlwu) [inline css/form.css:742]:
```suggestion
  color: #e03629;
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Rejected +1

File: js/components/file.js [hunk 2]
```diff
@@ -47,13 +47,18 @@ Fliplet.FormBuilder.field('file', {
   computed: {
     selectedFileName: function() {
       return _.map(this.value, 'name').join(', ');
+    },
+    isValueUrlLink: function() {
```
[the remark this answers is not in the document]
Comment (YaroslavOvdii) [reply js/components/file.js:51]: @tonytlwu You right we could use `Flipet.Media.isRemoteUrl()` I'll change it.

File: templates/components/file.build.hbs [hunk 3]
```diff
@@ -4,8 +4,8 @@
     <li class="file-item" v-for="(file, index) in value" @click="onFileItemClick(file.url)">
       <div class="file-title">\{{ file.name }}</div>
       <div class="file-info">
-        <span class="file-info-uploaded">Uploaded: \{{ showLocalDateFormat(file.createdAt) }}</span>
-        <span v-show="file.metadata && file.metadata.size" class="file-info-size">Size: \{{ humanFileSize(file.metadata && file.metadata.size) }}</span>
+        <span class="file-info-uploaded">Uploaded: <b>\{{ showLocalDateFormat(file.createdAt) }}</b></span>
```
[the remark this answers is not in the document]
Comment (YaroslavOvdii) [reply templates/components/file.build.hbs]: @tonytlwu Fixed

File: js/components/file.js [hunk 4]
```diff
@@ -74,7 +79,16 @@ Fliplet.FormBuilder.field('file', {
         files: fileIDs,
         fields: ['name', 'url', 'metadata', 'createdAt']
       }).then(function(files) {
-        $vm.value = files;
+        var newFiles = _.map(files, function(file) {
```
[the remark this answers is not in the document]
Comment (YaroslavOvdii) [reply js/components/file.js:82]: @tonytlwu Fixed

File: js/components/file.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (YaroslavOvdii) [reply js/components/file.js:51]: @tonytlwu Fixed

File: css/form.css (hunk 1 above)
Comment (YaroslavOvdii) [reply css/form.css]: @tonytlwu Fixed

File: css/form.css [hunk 5]
```diff
@@ -716,11 +712,43 @@ html[dir="rtl"] .radio.radio-icon input[type="radio"] + label > span.check {
...
+  max-width: calc(100% - 30px);
+}
+
+.file-holder.editable .file-icon {
+  width: 30px;
+  height: 30px;
+}
+
+.file-holder.editable .file-icon .fa {
+  color:#e03629;
```
Comment (tonytlwu) [inline css/form.css:742]:
```suggestion
  color: #e03629;
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Rejected +1

File: css/form.css (hunk 5 above)
Comment (YaroslavOvdii) [reply css/form.css]: @tonytlwu Fixed

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: High
Affected: All form components using the Files field
Verify: Users can load and upload files in forms

Squash and merge: Use PR title

**Pull request merged**